src/pymortests/algorithms/svd_va.py

```diff
@@ -31,14 +30,15 @@ def test_method_of_snapshots(vector_array, method):
 
     B = A.copy()
     with log_levels({"pymor.algorithms": "ERROR"}):
-        U, s, Vh = method(A)
+        U, s, Vh = method(A, rtol=4e-8)
     assert np.all(almost_equal(A, B))
     assert len(U) == len(s) == Vh.shape[0]
     assert Vh.shape[1] == len(A)
     assert np.allclose(Vh @ Vh.T.conj(), np.eye(len(s)))
-    U.scal(s)
-    UsVh = U.lincomb(Vh.T)
-    assert np.all(almost_equal(A, UsVh, rtol=4e-8))
+    if len(A) > 0:
```

Comment:
Does the rest actually fail for 0-length or is this just for efficiency?

Reply:
For 0-lenghts, `s[0]` does not exist.

Reply:
Right.

Reply:
Is it guaranteed that `len(s) > 0` when `len(A) > 0`? What if `A` is zero?

Reply:
To be on the safe side, I replaced `len(A)` by `len(s)` ...

Reply:
How about checking that `A` is zero when `len(s) == 0`?

Reply:
Does not have to be the case. Just choose your tolerances high enough.

Reply:
I mean for the case that is tested (`rtol=4e-8`). I would guess `len(s)==0` could only happen for nonzero `A` if `rtol>1`. But, maybe we don't have to test this very special case...